sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -261,9 +260,6 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
         val nameInCache = v2Ident.toQualifiedNameParts(catalog)
         isSameName(name, nameInCache, resolver) && (includeTimeTravel || timeTravelSpec.isEmpty)
 
-      case r: V2TableReference =>
```

Comment:
We only cache resolved plans, and we only perform cache lookup for resolved plans, so `V2TableReference` should never exist in the cache manager.